src/app/rmmapi/rmmauthguard.service.ts

```diff
@@ -58,18 +60,34 @@ export class RMMAuthGuardService implements CanActivate, CanActivateChild {
         });
     }
 
-    isLoggedIn(): Observable<boolean> {
+    isLoggedIn(): Observable<boolean | UrlTree> {
         return this.http.get('/rest/v1/me').pipe(
             map((res: any) => {
                 this.wasLoggedIn = res && res.status === 'success';
-                return this.wasLoggedIn;
+                if (res && res.result) {
+                    this.currentMe = res.result;
+                }
+                return this.checkStatus();
             }),
         );
     }
 
-    canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): boolean | Observable<boolean> | Promise<boolean> {
+    checkStatus(): boolean | UrlTree {
+        // Expired users are only allowed to visit account/subscriptions
+        if (this.currentMe && this.currentState
+            && this.currentMe.account_status === 'expired'
+            && this.currentState.url !== '/account/subscriptions') {
+            return this.router.parseUrl('/account/subscriptions');
+        } else {
+            return this.wasLoggedIn;
+        }
+        
+    }
+
+    canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): boolean | UrlTree | Observable<boolean | UrlTree> | Promise<boolean | UrlTree> {
         this.urlBeforeLogin = state.url;
 
+        this.currentState = state;
```

Comment:
Since you're only using the `.url` property of `RouterStateSnapshot`, wouldn't it be better to rename `currentState` to `currentURL` and change it to `this.currentURL = state.url`?

Reply:
Probably, guess I did future proof that a tad..

Reply:
I like to keep the least amount of state around that I need, saved me some headaches. Its up to you though, PR approved

Reply:
In fact your snippet just pointed out that the code is already saving the url .. so I've uselessly duplicated it! Yay for reviews

Reply:
Oh yea, lol, on the line right above it 😆